rmw_fastrtps_shared_cpp/src/rmw_request.cpp

```diff
@@ -63,6 +65,12 @@ __rmw_send_request(
     returnedValue = RMW_RET_OK;
     *sequence_id = ((int64_t)wparams.sample_identity().sequence_number().high) << 32 |
       wparams.sample_identity().sequence_number().low;
+    // This would ideally go before the write() call, but we can only get the sequence number after
```

Comment:
can we also add client gid as response data reader gid either with `info->reader_guid_` or calling `rmw_get_gid_for_client`? i am not sure why we are missing client gid explicitly here? is there any specific intension that we need to drop it? (the same comment goes to rmw_take_response)

Reply:
see https://github.com/ros2/ros2_tracing/pull/145#discussion_r1876858777